src/main/java/org/opensrp/domain/IdentifierSource.java

```diff
@@ -0,0 +1,151 @@
+package org.opensrp.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.opensrp.util.IdentifierValidatorAlgorithm;
+
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+
+public class IdentifierSource {
+
+	@JsonProperty
+	private Long id;
+
+	@JsonProperty
+	private String identifier;
+
+	@JsonProperty
+	private String description;
+
+	@Enumerated(EnumType.STRING)
+	@JsonProperty
+	private IdentifierValidatorAlgorithm identifierValidatorAlgorithm;
+
+	@JsonProperty
+	private String baseCharacterSet;
+
+	@JsonProperty
+	private String firstIdentifierBase;
+
+	@JsonProperty
+	private String prefix;
+
+	@JsonProperty
+	private String suffix;
+
+	@JsonProperty
+	private Integer minLength;
+
+	@JsonProperty
+	private Integer maxLength;
+
+	@JsonProperty
+	private String regexFormat;
```

Comment:
Can this be renamed as contains to have the regex that should be not generated.

Maybe name it `skipRegexFormat`

Reply:
Done

Reply:
Since we are not using the regex for skippable values, I think it should be named **regexFormat** @githengi what do you think?

Reply:
sure revert this to `regexFormat`

Reply:
Done